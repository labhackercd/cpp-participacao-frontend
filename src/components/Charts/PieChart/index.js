import React from 'react';
//import Plot from 'react-plotly.js';
import Chart from "react-google-charts";


export default function PieChart(props) {
  return (
      <Chart
        width={'500px'}
        height={'300px'}
        chartType="PieChart"
        loader={<div>Carregando Gr√°fico</div>}
        data={props.data}
        options={props.options}
        rootProps={{ 'data-testid': '1' }}
    />
   );
}

/*
export default function PieChart(props) {
  return (
        <Plot
        data={[
          {
            values: props.data.values,
            labels: props.data.labels,
            type: 'pie',
            marker: {
                colors: ['#855CF8', '#503795', '#E289F2']
            },
          },
        ]}
        layout={ { paper_bgcolor:"#000000", font: {color: '#ffffff'}} }
      />
   );
}*/
