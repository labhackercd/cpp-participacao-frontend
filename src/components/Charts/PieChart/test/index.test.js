import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from "enzyme";
import PieChart from './../index';

const pieChartdata = {
    values:[
        ['Dispositivo', 'Acessos'],
        ['Mobile', 15],
        ['Desktop', 15],
        ['Tablet', 70],
    ],
    options:{
        legend:{textStyle: {color: 'white'}},
        colors: ['#855CF8', '#503795', '#E289F2'],
        backgroundColor: '#000000',
    }
};

test("Snapshot should not have changes", () => {
    const component = mount(<PieChart data={pieChartdata.values} options={pieChartdata.options}></PieChart>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

it("Mounts the chart with the data", () => {
    const div = document.createElement("div")
    ReactDOM.render(<PieChart data={pieChartdata.values} options={pieChartdata.options}></PieChart>, div)

    /*
    const component = mount(<PieChart data={pieChartdata}></PieChart>);
    expect(component.exists()).toEqual(true);*/
});
