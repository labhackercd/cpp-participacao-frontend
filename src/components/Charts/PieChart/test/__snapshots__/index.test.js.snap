// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should not have changes 1`] = `
<PieChart
  data={
    Array [
      Array [
        "Dispositivo",
        "Acessos",
      ],
      Array [
        "Mobile",
        15,
      ],
      Array [
        "Desktop",
        15,
      ],
      Array [
        "Tablet",
        70,
      ],
    ]
  }
  options={
    Object {
      "backgroundColor": "#000000",
      "colors": Array [
        "#855CF8",
        "#503795",
        "#E289F2",
      ],
      "legend": Object {
        "textStyle": Object {
          "color": "white",
        },
      },
    }
  }
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="PieChart"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          "Dispositivo",
          "Acessos",
        ],
        Array [
          "Mobile",
          15,
        ],
        Array [
          "Desktop",
          15,
        ],
        Array [
          "Tablet",
          70,
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="300px"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div>
        Carregando Gráfico
      </div>
    }
    options={
      Object {
        "backgroundColor": "#000000",
        "colors": Array [
          "#855CF8",
          "#503795",
          "#E289F2",
        ],
        "legend": Object {
          "textStyle": Object {
            "color": "white",
          },
        },
      }
    }
    render={null}
    rootProps={
      Object {
        "data-testid": "1",
      }
    }
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="500px"
  >
    <ContextProvider
      value={
        Object {
          "chartActions": null,
          "chartEvents": null,
          "chartType": "PieChart",
          "chartWrapperParams": Object {},
          "className": "",
          "columns": null,
          "controls": null,
          "data": Array [
            Array [
              "Dispositivo",
              "Acessos",
            ],
            Array [
              "Mobile",
              15,
            ],
            Array [
              "Desktop",
              15,
            ],
            Array [
              "Tablet",
              70,
            ],
          ],
          "diffdata": null,
          "formatters": null,
          "getChartEditor": null,
          "getChartWrapper": [Function],
          "graphID": null,
          "graph_id": null,
          "height": "300px",
          "legendToggle": false,
          "legend_toggle": false,
          "loader": <div>
            Carregando Gráfico
          </div>,
          "options": Object {
            "backgroundColor": "#000000",
            "colors": Array [
              "#855CF8",
              "#503795",
              "#E289F2",
            ],
            "legend": Object {
              "textStyle": Object {
                "color": "white",
              },
            },
          },
          "render": null,
          "rootProps": Object {
            "data-testid": "1",
          },
          "rows": null,
          "spreadSheetQueryParameters": Object {
            "gid": 1,
            "headers": 1,
          },
          "spreadSheetUrl": null,
          "style": Object {},
          "toolbarID": null,
          "toolbarItems": null,
          "width": "500px",
        }
      }
    >
      <div>
        Carregando Gráfico
      </div>
      <GoogleChartLoader
        onError={[Function]}
        onLoad={[Function]}
      >
        <Script
          attributes={Object {}}
          onCreate={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          url="https://www.gstatic.com/charts/loader.js"
        />
      </GoogleChartLoader>
    </ContextProvider>
  </Chart>
</PieChart>
`;
